docs/src/math_framework.md

```diff
@@ -74,6 +74,58 @@ while the _contravariant basis_ is the opposite: gradient in ``x`` of the coordi
 \mathbf{e}^i = \nabla_x \xi^i
 ```
 
+Here is an intuitive example of how the _contravariant basis_ and _covariant basis_ changes when a coordinate is being transformed.
+
+![Different bases supported in ClimaCore.jl](controva_cova.png)
+
+Start with the _contravariant basis_, which is the case that most of us is more familier. As it can be seen in the figure below, we have 2D vector $\vec{a}$, 
+and two unit vectors $\vec{e_{1}}$ and $\vec{e_{2}}$, served as two basis. Thus we can represent vector $\vec{a}$ as:
+```math
+\vec{a} = a^{1} \vec{e_{1}} + a^{2} \vec{e_{2}}
+```
+
+And now we set amplify one of the basis $\vec{a}$ by 2 times, that is $\vec{e_{1}}^{'} = 2\vec{e_{1}}$. If we want to maintain the vector $\vec{a}$ to stay still, 
+obviously we need to set $a^{1}$ to its half:
+```math
+a^{1'}  = \frac{1}{2} a^{1}
+```
+
+With this being said, the components we use to describe a certain vector are changing in the opposite manner of the the basis, and we are calling these 
+components _contravariant components_. The _contravariant vector_ is a vector with _contravariant components_. 
```

Comment:
This does not match what @dennisYatunin  said in the PR conversation. 

> `CovariantVectors` are aligned with the contravariant basis vectors, but have covariant components

Please update accordingly